compiler/src/iree/compiler/Dialect/Util/IR/UtilAttrs.cpp

```diff
@@ -208,51 +221,111 @@ static LogicalResult serializeRawData(DenseElementsAttr elementsAttr,
   return success();
 }
 
-template <typename elementType, unsigned numBits = sizeof(elementType) * 8>
-static LogicalResult
-serializeGenericIntElements(DenseIntElementsAttr attr,
-                            llvm::support::endianness endian,
-                            llvm::raw_ostream &os) {
-  for (const APInt &value : attr.getValues<APInt>()) {
-    elementType rawValue = llvm::support::endian::byte_swap<elementType>(
-        value.extractBitsAsZExtValue(numBits, 0), endian);
-    os.write((char *)&rawValue, sizeof(rawValue));
+template <typename physicalType,
+          unsigned physicalBitWidth = sizeof(physicalType) * 8>
+class PackedWriter {
```

Comment:
Put some comments around how this works? Esp. around endianness and such, so it's easier to understand later. :)